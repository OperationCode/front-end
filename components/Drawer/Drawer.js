import React from 'react';
import { node, string, bool } from 'prop-types';
import classNames from 'classnames';
import styles from './Drawer.module.css';

Drawer.propTypes = {
  children: node.isRequired,
  className: string,
  isVisible: bool,
};

Drawer.defaultProps = {
  className: undefined,
  isVisible: false,
};

function Drawer({ children, className, isVisible }) {
  return (
    <div
      className={classNames(className, {
        [styles.visible]: isVisible,
        [styles.hidden]: !isVisible,
      })}
    >
      <div className={styles.content}>{children}</div>
    </div>
  );
}

export default Drawer;
