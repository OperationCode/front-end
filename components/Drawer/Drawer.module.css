.body {
  display: none;
  position: fixed;
  top: 0;
  bottom: 0;
  overflow: hidden;
  width: 100%;
  z-index: 2;
  transition: left 0.25s ease-in-out;
}

.hidden {
  composes: body;
  left: -100%;
}

.visible {
  composes: body;
  left: 0;
}

.content {
  height: 100%;
  width: 100%;
}

@media screen and (--large-viewport) {
  .body {
    display: block;
  }
}

@media screen and (orientation: landscape) {
  .body {
    width: 320px;
  }
}
