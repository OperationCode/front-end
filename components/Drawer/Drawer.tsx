import classNames from 'classnames';

export interface DrawerPropsType {
  /**
   * Content to be rendered in the Drawer.
   */
  children: React.ReactNode;
  /**
   * Applies class names to the container element.
   */
  className?: string;
  /**
   * Sets if the content is visible.
   * @default false
   */
  isVisible?: boolean;
}

export function Drawer({ children, className, isVisible = false }: DrawerPropsType) {
  return (
    <div
      className={classNames(
        className,
        'hidden lg:block transition-all ease-in-out duration-1000 fixed top-0 bottom-0  overflow-hidden width-full z-[2]',
        {
          '-left-0': isVisible,
          '-left-full': !isVisible,
        },
      )}
    >
      <div className="w-full h-full">{children}</div>
    </div>
  );
}
