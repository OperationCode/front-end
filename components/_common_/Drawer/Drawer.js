import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Drawer.css';

Drawer.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isVisible: PropTypes.bool,
};

Drawer.defaultProps = {
  className: '',
  isVisible: false,
};

function Drawer({ children, className, isVisible }) {
  return (
    <div
      className={classNames(className, {
        [styles.visible]: isVisible,
        [styles.hidden]: !isVisible,
      })}
    >
      <div className={styles.content}>{children}</div>
    </div>
  );
}

export default Drawer;
