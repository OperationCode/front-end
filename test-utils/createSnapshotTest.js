/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
