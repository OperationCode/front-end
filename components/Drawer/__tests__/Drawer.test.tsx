import { Drawer } from '../Drawer';
import createSnapshotTest from '@/test-utils/createSnapshotTest';

describe('Drawer', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Drawer>Test</Drawer>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Drawer className="test-class" isVisible>
        Test
      </Drawer>,
    );
  });
});
