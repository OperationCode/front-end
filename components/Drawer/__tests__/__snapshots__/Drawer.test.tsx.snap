// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Drawer > should render with many props assigned 1`] = `
<div
  className="test-class hidden lg:block transition-all ease-in-out duration-1000 fixed top-0 bottom-0  overflow-hidden width-full z-[2] -left-0"
>
  <div
    className="w-full h-full"
  >
    Test
  </div>
</div>
`;

exports[`Drawer > should render with required props 1`] = `
<div
  className="hidden lg:block transition-all ease-in-out duration-1000 fixed top-0 bottom-0  overflow-hidden width-full z-[2] -left-full"
>
  <div
    className="w-full h-full"
  >
    Test
  </div>
</div>
`;
